casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -569,26 +551,8 @@ object Validate {
       previousBlockJustificationHash: BlockHash
   ): F[Boolean] =
     for {
-      maybeCurrentBlockJustification <- dag.lookup(currentBlockJustificationHash)
-      currentBlockJustification <- maybeCurrentBlockJustification match {
-                                    case Some(block) => block.pure[F]
-                                    case None =>
-                                      Sync[F].raiseError[BlockMetadata](
-                                        new Exception(
-                                          s"Missing ${PrettyPrinter.buildString(currentBlockJustificationHash)} from block dag store."
-                                        )
-                                      )
-                                  }
-      maybePreviousBlockJustification <- dag.lookup(previousBlockJustificationHash)
-      previousBlockJustification <- maybePreviousBlockJustification match {
-                                     case Some(block) => block.pure[F]
-                                     case None =>
-                                       Sync[F].raiseError[BlockMetadata](
-                                         new Exception(
-                                           s"Missing ${PrettyPrinter.buildString(previousBlockJustificationHash)} from block dag store."
-                                         )
-                                       )
-                                   }
+      currentBlockJustification  <- dag.unsafeLookup(currentBlockJustificationHash)
+      previousBlockJustification <- dag.unsafeLookup(previousBlockJustificationHash)
```

Comment:
The function unsafeLookup seems to be a way to remove this duplication. If that's the case, I don't think it's a good idea. You could easily have a function with a signature like this to eliminate the duplication:
```scala
def failOnNone : F[Option[A]] => F[A]
```

Reply:
I prefer hiding the fact that it can throw an exception as an implementation detail

Reply:
That is never an implementation detail. As soon as you do that you've got no local reasoning and no types can help you. I second @odeac's remarks. No more `unsafe*` stuff should get into our codebase.

Reply:
Sorry I got mislead by the name. I thought we're talking about an effectful function not returning via `F[_]`. I think it's fine to not parameterize with the exception. The name should be changed though, as it causes an instant trigger :) Maybe `tryLookup`?